libs/SalesforceSDK/src/com/salesforce/androidsdk/auth/NativeLoginManager.kt

```diff
@@ -430,23 +512,78 @@ internal class NativeLoginManager(
         )
     }
 
+    @Deprecated(
+        "The replacement function provided in 12.1.0 uses a more consistent naming convention.",
+        ReplaceWith("submitCompletePasswordLessAuthorizationRequest(otp, otpIdentifier, otpVerificationMethod)")
+    )
+    override suspend fun submitPasswordlessAuthorizationRequest(
+        otp: String,
+        otpIdentifier: String,
+        otpVerificationMethod: OtpVerificationMethod
+    ) = completePasswordLessAuthorization(
+        otp,
+        otpIdentifier,
+        otpVerificationMethod
+    )
+
     /**
-     * A data class for the OTP request body.
+     * A data class for the start password reset request body.
      * @param recaptcha The reCAPTCHA token provided by the reCAPTCHA Android
      * SDK.  This is not used with reCAPTCHA Enterprise
      * @param recaptchaevent The reCAPTCHA parameters for use with reCAPTCHA
      * Enterprise
-     * @param username The Salesforce username
+     * @param username A valid Salesforce username.  Note that email may be used
+     * for community users
+     */
+    private data class StartPasswordResetRequestBody(
+        val recaptcha: String?,
+        val recaptchaevent: ReCaptchaEventRequestParameter?,
+        val username: String
+    ) {
+        fun toJson() = JSONObject().apply {
+            put("recaptcha", recaptcha)
+            put("recaptchaevent", recaptchaevent?.toJson())
+            put("username", username)
+        }
+    }
+
+    /**
+     * A data class for the complete reset password OTP request body.
+     * @param username A valid Salesforce username.  Note that email may be used
+     * for community users
+     * @param otp The user-entered one-time-password previously delivered to the
+     * user by the Salesforce Identity API forgot password endpoint
+     * @param newPassword The user-entered new password
+     */
+    private data class CompletePasswordResetRequestBody(
+        val username: String,
+        val otp: String,
+        val newPassword: String
+    ) {
+        fun toJson() = JSONObject().apply {
+            put("username", username)
+            put("otp", otp)
+            put("newpassword", newPassword)
+        }
+    }
+
+    /**
+     * A data class for the start password-less login request body.
+     * @param recaptcha The reCAPTCHA token provided by the reCAPTCHA Android
+     * SDK.  This is not used with reCAPTCHA Enterprise
+     * @param recaptchaevent The reCAPTCHA parameters for use with reCAPTCHA
+     * Enterprise
+     * @param username A valid Salesforce username.  Note that email may be used
+     * for community users
      * @param verificationMethod The OTP verification code's delivery method in
      * "email" or "sms"
      */
-    private data class OtpRequestBody(
+    private data class StartPasswordLessLoginRequestBody(
```

Comment:
In favor of renaming the password-less login methods and data classes, there's not a generic "OTP Request" in the API.  Each endpoint and step in the flow has a distinct object graph.